components/eventing-controller/controllers/subscription/jetstream/reconciler.go

```diff
@@ -140,6 +140,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	// Check for valid sink
 	if err := r.sinkValidator.Validate(desiredSubscription); err != nil {
+		if deleteErr := r.Backend.DeleteSubscriptionsOnly(desiredSubscription); deleteErr != nil {
```

Comment:
I think the invalid subscription deletion is a part of syncing the Subscription to JetStream, which is `r.Backend.SyncSubscription(desiredSubscription)`. It'd more clean if it is part of that function.

Reply:
We don't reach `r.Backend.SyncSubscription(desiredSubscription)` because the `r.sinkValidator.Validate` returns an error before it.

Reply:
I know, but it is logically part of `SyncSubscription`. That is, whenever a Subscription is changed the function syncs the event types to the JetStream backend. Ideally, validation needs to be called in side that function where Subscription can be deleted & updated.
Let's keep the change minimal as you did, acceptable for me.